src/Assets/LightSkyDefense/Scripts/SpawnCreditOnDestruction.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpawnCreditOnDestruction : MonoBehaviour
+{
+    public Credit credit;
+    public int value;
+
+    private bool _isQuitting = false;
+
+    void OnApplicationQuit()
+    {
+        _isQuitting = true;
+    }
+
+    /// <summary>
+    /// Spawn money when its gameObject is destroyed.
+    /// </summary>
+    void OnDestroy()
+    {
+        if (!_isQuitting)
```

Comment:
You can invert this if to avoid nesting.

```suggestion
        if (_isQuitting) return;

```